velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -807,6 +807,34 @@ struct MakeYMIntervalFunction {
   }
 };
 
+template <typename T>
+struct MakeDTIntervalFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<IntervalDayTime>& result,
+      int32_t days = 0,
+      int32_t hours = 0,
+      int32_t minutes = 0,
+      double seconds = 0) {
+
+    auto totalMicros = (int64_t)days * Timestamp::kSecondsInDay * Timestamp::kMicrosecondsInSecond +
+                       (int64_t)hours * 3600 * Timestamp::kMicrosecondsInSecond +
+                       (int64_t)minutes * 60 * Timestamp::kMicrosecondsInSecond +
+                       (int64_t)seconds * Timestamp::kMicrosecondsInSecond;
```

Comment:
Overflow can already happen during one of these operations before the check. Perhaps use safe arithmetic functions including `__builtin_mul_overflow`  and `__builtin_add_overflow` to detect overflow during calculations.